metricbeat/docs/fields.asciidoc

```diff
@@ -9617,309 +9627,827 @@ Number of rollovers assertions produced by the server.
+Reports on lock state of the database.
 
 
 
-*`mongodb.status.connections.current`*::
+*`mongodb.status.global_lock.total_time.us`*::
 +
 --
 type: long
 
-The number of connections to the database server from clients. This number includes the current shell session. Consider the value of `available` to add more context to this datum.
+The time, in microseconds, since the database last started and created the globalLock. This is roughly equivalent to total server uptime.
 
 
 --
 
-*`mongodb.status.connections.available`*::
+[float]
+== current_queue fields
+
+The number of operations queued because of a lock.
+
+
+
+*`mongodb.status.global_lock.current_queue.total`*::
 +
 --
 type: long
 
-The number of unused available incoming connections the database can provide.
+--
 
+*`mongodb.status.global_lock.current_queue.readers`*::
++
+--
+type: long
 
 --
 
-*`mongodb.status.connections.total_created`*::
+*`mongodb.status.global_lock.current_queue.writers`*::
 +
 --
 type: long
 
-A count of all incoming connections created to the server. This number includes connections that have since closed.
+--
+
+[float]
+== active_clients fields
+
+The number of connected clients and the read and write operations performed by these clients.
+
+
+
+*`mongodb.status.global_lock.active_clients.total`*::
++
+--
+type: long
+
+--
+
+*`mongodb.status.global_lock.active_clients.readers`*::
++
+--
+type: long
+
+The number of the active client connections performing read operations.
+
+
+--
+
+*`mongodb.status.global_lock.active_clients.writers`*::
++
+--
+type: long
+
+The number of the active client connections performing write operations.
 
 
 --
 
 [float]
-== journaling fields
+== locks fields
 
-Data about the journaling-related operations and performance. Journaling information only appears for mongod instances that use the MMAPv1 storage engine and have journaling enabled.
+A document that reports for each lock <type>, data on lock <mode>s. The possible lock <type>s are global, database, collection, metadata and oplog. The possible <mode>s are r, w, R and W which respresent shared, exclusive, intent shared and intent exclusive.
+locks.<type>.acquire.count.<mode> shows the number of times the lock was acquired in the specified mode. locks.<type>.wait.count.<mode> shows the number of times the locks.acquireCount lock acquisitions encountered waits because the locks were held in a conflicting mode. locks.<type>.wait.us.<mode> shows the cumulative wait time in microseconds for the lock acquisitions. locks.<type>.deadlock.count.<mode> shows the number of times the lock acquisitions encountered deadlocks.
 
 
 
-*`mongodb.status.journaling.commits`*::
+
+*`mongodb.status.locks.global.acquire.count.r`*::
 +
 --
 type: long
 
-The number of transactions written to the journal during the last journal group commit interval.
+--
 
+*`mongodb.status.locks.global.acquire.count.w`*::
++
+--
+type: long
 
```

Comment:
Could you add a description for all these new fields?

Reply:
For the fields under the `locks`, there is a general description under the `locks`:
```
A document that reports for each lock <type>, data on lock <mode>s.
The possible lock <type>s are global, database, collection, metadata, and oplog.
The possible <mode>s are r, w, R and W which represents shared, exclusive, intent shared and intent exclusive.

locks.<type>.acquire.count.<mode> shows the number of times the lock was acquired in the specified mode.
locks.<type>.wait.count.<mode> shows the number of times the locks.acquireCount lock acquisitions encountered wait because the locks were held in a conflicting mode.
locks.<type>.wait.us.<mode> shows the cumulative wait time in microseconds for the lock acquisitions.
locks.<type>.deadlock.count.<mode> shows the number of times the lock acquisitions encountered deadlocks.
```
Considering this description, should I add a description for every field?

Reply:
Ok, it's fine, we can revisit it in the future if needed.